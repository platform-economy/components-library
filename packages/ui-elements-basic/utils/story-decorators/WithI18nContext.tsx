import * as React from 'react';
// @ts-ignore
import { withI18n } from 'react-i18next';

export interface WithI18nContextProps {
  /** Children of context decorator */
  children: React.ReactDOM;
}

const WithI18nContext = withI18n()((props: WithI18nContextProps) => (<div>{props.children}</div>));

export function withI18nContext(story: () => unknown) {
  return (
    <WithI18nContext>
      {story()}
    </WithI18nContext>
  );
}
