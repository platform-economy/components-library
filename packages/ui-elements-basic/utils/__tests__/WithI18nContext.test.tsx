import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';

import { withI18nContext } from '../story-decorators/WithI18nContext';

describe('ui-elements-core', () => {
  describe('utils', () => {
    describe('WithI18nContext', () => {

      it('should render component with I18N context correctly', () => {
        const tree = TestRenderer
          .create(withI18nContext(() => (<div>Test Component</div>)))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

    });
  });
});
