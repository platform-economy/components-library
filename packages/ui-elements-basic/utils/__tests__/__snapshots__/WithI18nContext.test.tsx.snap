// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-core utils WithI18nContext should render component with I18N context correctly 1`] = `
<div>
  <div>
    Test Component
  </div>
</div>
`;
